run.sh

```diff
@@ -13,15 +20,206 @@ WP_DEBUG=${WP_DEBUG:-'false'}
   AFTER_URL=$(echo "$SEARCH_REPLACE" | cut -d ',' -f 2) || \
   SEARCH_REPLACE=false
 
-ERROR () {
-  echo -e "\n=> $(tput -T xterm setaf 1)$(tput -T xterm bold)ERROR$(tput -T xterm sgr 0) (Line $1): $2.";
-  exit 1;
+
+main() {
+
+  generate_config_for wp-cli
+
+  # Download WordPress
+  # ------------------
+  if [ ! -f /app/wp-settings.php ]; then
+    h3 "Downloading wordpress"
+    chown -R www-data:www-data /app /var/www/html
+    WP core download |& loglevel
+    STATUS ${PIPESTATUS[0]}
+  fi
+
+  # Wait for MySQL
+  # --------------
+  h2 "Waiting for MySQL to initialize..."
+  while ! mysqladmin ping --host=db --password=$DB_PASS --silent; do
+    sleep 1
+  done
+
+  h1 "Begin WordPress Configuration"
+
+  h3 "Generating wp.config.php file"
+  generate_config_for wordpress
+  STATUS $?
+
+  h2 "Checking database"
+  check_database
+
+  # .htaccess
+  # ---------
+  if [ ! -f /app/.htaccess ]; then
+    h3 "Generating .htaccess file"
+    if [[ "$MULTISITE" == 'true' ]]; then
+      STATUS 1
+      h3warn "Cannot generate .htaccess for multisite!"
+    else
+      WP rewrite flush --hard |& loglevel
+      STATUS ${PIPESTATUS[0]}
+    fi
+  else
+    h3 ".htaccess exists... SKIPPING"
+    STATUS SKIP
+  fi
+
+  h3 "Adjusting filesystem permissions"
+  groupadd -f docker && usermod -aG docker www-data
+  find /app -type d -exec chmod 755 {} \;
+  find /app -type f -exec chmod 644 {} \;
+  mkdir -p /app/wp-content/uploads
+  chmod -R 775 /app/wp-content/uploads && \
+    chown -R :docker /app/wp-content/uploads
+  STATUS $?
+
+  h2 "Checking plugins"
+  check_plugins
+
+  # Make multisite
+  # ---------------
+  h2 "Checking for WordPress Multisite..."
+  if [ "$MULTISITE" == "true" ]; then
+    h3 "Multisite found. Enabling..."
+    WP core multisite-convert |& loglevel
+    STATUS ${PIPESTATUS[0]}
+  else
+    h3 "Multisite not found. Skipping..."
+    STATUS SKIP
+  fi
+
+  # Operations to perform on first build
+  # ------------------------------------
+  if [ -d /app/wp-content/plugins/akismet ]; then
```

Comment:
This is not reliable, some install need `akismet`. We could `touch .first_install` or similar